services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,32 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  xQueueSend(thermalMgrQueueHandle, event, portMAX_DELAY);
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event; event.type = THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK;
+  xQueueSend(thermalMgrQueueHandle, &event, portMAX_DELAY);
+  
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t *config = (void *) pvParameters;
+  thermal_mgr_event_t event;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+      float temp;
+      readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
In general, always check the return value of non void functions for error codes

Reply:
Also check the event before reading temp. If the event is not valid (i.e. not check temp or THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK) we should not do anything